README.md

```diff
@@ -146,18 +200,60 @@ MessageAssert assertions = BpmnAssert.assertThat(response);
 >
 > Any feedback about the wait for idle state is highly appreciated! Please let us know if the delay should be higher, or configurable.
 
-The engine allows you to wait until it is idle before continuing with your test.
+`engine.waitForIdleState(timeout)` will cause your test to stop executing until the engine has
+reached an idle state. If the engine does not reach an idle state within the specified timeout a
+`TimeoutException` will be thrown.
+
+We have defined an idle state as a state in which the engine makes no progress and is waiting for
+new commands or events to trigger. Once the engine has detected it has become idle it will wait for
+a delay (30ms) and check if it is still idle. If this is the case it is considered to be in idle
+state and continue your test.
+
+#### Wait for busy state
 
-`engine.waitForIdleState()` - This method will cause your test to stop executing until the engine has reached the idle state.
+`engine.waitForBusyState(timeout)` will cause your test to stop executing until the engine has
+reached a busy state. If the engine does not reach a busy state within the specified timeout a
+`TimeoutException` will be thrown.
 
-We have defined an idle state as a state in which the process engine makes no progress and is waiting for new commands or events to trigger.
-Once the engine has detected it has become idle it will wait for a delay (30ms) and check if it is still idle.
-If this is the case it is considered to be in idle state and continue your test / execute the runnables.
+We consider the engine to have reached a busy state when any new record / command is processed since
+we've started waiting.
+
+Waiting for a busy state is useful in scenarios where you're expecting the engine to start doing
+something, without explicitly triggering it yourself. An example of this would be a process with a
+timer event. We can increase the time of the engine, but we cannot trigger the timer explicitly.
+Because of this we should wait for a busy state after increasing the engine time.
 
 ## Examples
 
 For example tests the best place to look right now is the tests in the QA module.
 
+## Switching between testcontainers and embedded
+
+Switching between testcontainers and embedded is very easy to do. You'll have to take two steps:
+
+1. Switch to the relevant dependency
+   - Testcontainers: `zeebe-process-test-extension-testcontainer`
+   - Embedded: `zeebe-process-test-extension`
+2. Change the import of `@ZeebeProcessTest`
+   - Testcontainers: `import io.camunda.zeebe.process.test.extension.testcontainer.ZeebeProcessTest;`
+   - Embedded: `import io.camunda.zeebe.process.test.extension.ZeebeProcessTest;`
+
+## Engine lifecycle
+
+The lifecycle of the engine will be fully managed by the extension. The lifecycle for both
+extensions differs slightly.
+
+**Testcontainers**
+1. Before all tests start the testcontainer
```

Comment:
The scope of all tests isn't fully clear. Is it a test class, a nested class, a test suite?